interpreter/k8s/src/main/scala/io/buoyant/interpreter/k8s/ConfigMapInterpreterInitializer.scala

```diff
@@ -41,48 +50,63 @@ case class ConfigMapInterpreterConfig(
   private[this] val log = Logger()
 
   @JsonIgnore
-  val api = {
-    val client = mkClient(Params.empty).configured(Label("configMapInterpreter"))
-    val nsOrDefault = namespace.getOrElse(DefaultNamespace)
-    Api(client.newService(dst)).withNamespace(nsOrDefault)
-  }
+  val api = mkApi.getOrElse(mkDefaultApi)()
 
   @JsonIgnore
   val act = api.configMap(name)
     .activity(extractDtab) {
-      (dtab, event) =>
+      (oldDtab, event) =>
         event match {
-          case ConfigMapAdded(a) => getDtab(a)
-          case ConfigMapModified(m) => getDtab(m)
+          // TODO: if the event doesn't update our dtab, don't update the activity?
+          case ConfigMapDtab(Return(newDtab)) => newDtab
+          case ConfigMapDtab(Throw(e)) =>
+            log.error("k8s ConfigMap %s error reading dtab '%s': %s", name, filename, e)
+            oldDtab
           case ConfigMapDeleted(_) =>
             log.warning(s"k8s ConfigMap $name was deleted!")
             Dtab.empty
           case ConfigMapError(e) =>
             log.error("k8s watch error: %s", e)
-            dtab
+            oldDtab
         }
     }
 
   @JsonIgnore
   @inline
   private[this] def extractDtab(response: Option[ConfigMap]): Dtab =
-    response.map(getDtab).getOrElse {
-      log.warning(s"K8s ConfigMap %s doesn't exist, assuming it will be created", name)
-      Dtab.empty
+    response.map(getDtab) match {
+      case Some(Return(dtab)) => dtab
+      case Some(Throw(e)) =>
+        log.error("k8s ConfigMap %s error reading dtab '%s': %s", name, filename, e)
+        Dtab.empty
+      case None =>
+        log.warning(s"K8s ConfigMap %s doesn't exist, assuming it will be created", name)
+        Dtab.empty
     }
 
   @JsonIgnore
-  def getDtab(configMap: ConfigMap): Dtab =
+  def getDtab(configMap: ConfigMap): Try[Dtab] =
     configMap.data.get(filename) match {
       case None =>
         log.warning(s"dtab at %s in k8s ConfigMap %s did not exist!", filename, name)
-        Dtab.empty
-      case Some(data) => Dtab.read(data)
+        Return(Dtab.empty)
+      case Some(data) => Try(Dtab.read(data))
```

Comment:
TIOLI: I think you could leave the existing signature the same if you changed this to

```
Try(Dtab.read(data)).getOrElse(Dtab.empty)
```

or

```
try { Dtab.read(data) } catch { case _: IllegalArgumentException =>
  log.warning(...)
  Dtab.empty
}
```

Reply:
Hmm, this changes the semantics somewhat: if we want to fall back to the previous dtab on an error, rather than returning `Dtab.empty`, the error needs to at least propagate up to `Activity.onEvent`, where the previous dtab value is known...